tests/webTables.spec.ts

```diff
@@ -1,222 +1,133 @@
+            .filter({has: page.getByRole('cell').nth(2)})).toHaveCount(4)
+    
+    });
+    
+    test('Validate search by Last Name', async ({ page }) => {
+        /* 2. - 8. In the "Last name" input field, type the last name "Black", "Davis", "Es" (partial string), and "Playwright". */
+        // Search for the following last names and validate the results
+        const lastNames = ['Black', 'Davis', 'Es', 'Playwright']
+    
+        for(const lastName of lastNames){
+            // Use the search field to locate entries with a specific last name
+            await page.getByRole('textbox').fill(lastName)
+            await page.getByText('Find Owner').click()
+            
+            // Capture the rows in the table body (not the header)
+            const rows = await page.locator('tbody tr').all()
+    
+            // For each resulting row, check for the last name
+            for(const row of rows){
+    
+                // Ensuring the resulting row has cells. There is an extra row in the 'Pets' column that we need to filter out, as it contains no cells.
+                if((await row.getByRole('cell').all()).length > 0){
+    
+                    // Checking if the name isn't our negative test, 'Playwright'.
+                    if(lastName !== 'Playwright'){
+    
+                        // Validating that the last name is in the first cell in the row
+                        await expect(row.getByRole('cell').first()).toContainText(lastName) 
+                    }
+                    else{
+                        /* 9. Add the assertion of the message "No owners with LastName starting with "Playwright"" */
+                        // Validate that the error message is correct for a last name that cannot be found
+                        await expect(page.locator('h2').locator('..').locator('div').last()).toHaveText("No owners with LastName starting with \"Playwright\"")
+                    }
+                }
             }
         }
-    }
-});
-
-test('Validate pets of the Madison city', async ({ page }) => {
-    // Navigate to the owners table
-    await page.getByText('Owners').click()
-    await page.getByText('Search').click()
-    // Wait for all table contents to load
-    await page.waitForTimeout(timeout)
-
-    const expectedPetsList = ['Freddy', 'George', 'Leo', 'Mulligan']
-    let actualPetsList: string[] = []
+    });
+    
+    test('Validate phone number and pet name on the Owner Information page', async ({ page }) => {
+        /* 2. Locate the owner by the phone number "6085552765". Extract the Pet name displayed in the table for the owner and save it to the variable. Click on this owner. */
+        const phoneNumber = '6085552765'
+        
+        // Capture the row with the above associated phone number
+        const row = page.getByRole('row', {name: phoneNumber})
+        
+        // Extract the pet name from the row
+        const petName = await row.locator('td').nth(4).textContent()!
```

Comment:
Exclamation sign should be not here.
It should be at the end of the argument like `petName!` when you use it later

Reply:
Ok, thank you!